src/app/shell/personal-cabinet/provider/provider-study-subjects/provider-study-subjects.component.ts

```
@@ -151,28 +138,68 @@ export class ProviderStudySubjectsComponent extends ProviderComponent implements
 
     this.getStudySubjects();
   }
 
-  public onDateApply(): void {
-    this.filterForm.patchValue({ dateFrom: this.tempDateFrom, dateTo: this.tempDateTo });
-    this.setDateForFilters();
+  public getStudySubjects(): void {
+    const from = this.datesGroup?.get('dateFrom')?.value;
+    const to = this.datesGroup?.get('dateTo')?.value;
+
+    this.subjectParameters.dateFrom = from && from.isValid() ? from?.format('YYYY-MM-DD') : '';
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)